frontend/public/components/custom-resource-definition.tsx

```diff
@@ -107,6 +123,68 @@ const Established: React.FC<{ crd: CustomResourceDefinitionKind }> = ({ crd }) =
   );
 };
 
+const EmptyVersionsMsg: React.FC<{}> = () => <EmptyBox label="CRD Versions" />;
+
+const CRDVersionsHeader = [
+  {
+    title: 'Name',
+    transforms: [sortable],
+  },
+  {
+    title: 'Served',
+    transforms: [sortable],
+  },
+  {
+    title: 'Storage',
+    transforms: [sortable],
+  },
+];
+
+const CRDVersionsRows = (versions: CRDVersion[]) =>
+  _.map(versions, (version: CRDVersion) => [
+    version.name,
+    version.served.toString(),
+    version.storage.toString(),
+  ]);
+
+const CRDVersionTable: React.FC<CRDVersionProps> = ({ versions }) => {
+  const [versionRows, setVersionRows] = React.useState<string[][]>(CRDVersionsRows(versions));
```

Comment:
Don't store this in state, just sort each render. The component should only re-render when the values change or sort by changes.